bot/exts/backend/sync/_syncers.py

```diff
@@ -183,12 +200,47 @@ async def _get_diff(self, guild: Guild) -> _Diff:
 
         return _Diff(users_to_create, users_to_update, None)
 
+    async def _get_users(self, endpoint: str = "bot/users", query_params: list = None) -> t.List[dict]:
+        """GET all users recursively."""
+        users = []
+        response: dict = await self.bot.api_client.get(endpoint, params=query_params)
+        users.extend(response["results"])
+
+        # The `response` is paginated, hence check if next page exists.
+        if (next_page_url := response["next"]) is not None:
+            next_endpoint, query_params = self.get_endpoint(next_page_url)
+            users.extend(await self._get_users(next_endpoint, query_params))
+        return users
+
+    @staticmethod
+    def get_endpoint(url: str) -> t.Tuple[str, t.List[tuple]]:
+        """Extract the API endpoint and query params from a URL."""
+        url = urlparse(url)
+
+        # Do not include starting `/` for endpoint.
+        endpoint = url.path[1:]
+
+        # Query params.
+        params = parse_qsl(url.query)
+
+        return endpoint, params
+
+    @staticmethod
+    def patch_dict(user: _User) -> t.Dict[str, t.Union[int, str, tuple, bool]]:
+        """Convert namedtuple to dict by omitting None values."""
+        user_dict = {}
+        for field in user._fields:
+            if (value := getattr(user, field)) is not None:
+                user_dict[field] = value
+        return user_dict
+
     async def _sync(self, diff: _Diff) -> None:
         """Synchronise the database with the user cache of `guild`."""
         log.trace("Syncing created users...")
-        for user in diff.created:
-            await self.bot.api_client.post('bot/users', json=user._asdict())
-
+        if diff.created:
+            created = [user._asdict() for user in diff.created]
+            await self.bot.api_client.post("bot/users", json=created)
```

Comment:
Leave some space between these to make it more readable.

```suggestion
            await self.bot.api_client.post("bot/users", json=created)
            
```